validation/validate_dc2_run2.2i_object_table.ipynb

```diff
@@ -502,8 +502,10 @@
     "else:\n",
     "    print(f'For {catalog_name} with {sampling_factor}x subsample')\n",
     "\n",
-    "if restrict_to_tracts:\n",
-    "    print(f'Tracts: ', restrict_to_tracts)"
+    "try:\n",
+    "    print(f'Tracts: ', restrict_to_tracts)\n",
+    "except:\n",
```

Comment:
> First, I think when LOCAL_DASK is set to False, this notebook only works when using an "Exclusive CPU Node". If I use "Shared CPU Node", I was not able to connect to the dask workers that I set up following the instructions.

I'm happy to make this note, but it hadn't occurred to me to ask for a shared CPU Node.  The example `salloc` command should get you exclusive nodes.  Did you get "Shared CPU Nodes" because you specifically asked for them?

Reply:
> Then, once I successfully connect to the dask client and run a few cell, I quickly ran into memory issue. In particular, it says there's not enough memory and then killed the dask workers. Some example error message looks like:

Was this with the suggested configuration of 32 total workers across two Haswell nodes?

Reply:
Sorry for my unclear statement, I was talking about the options of NERSC jupyter hub:

![image](https://user-images.githubusercontent.com/3792659/87969298-db80b200-ca8f-11ea-8e97-3604936db8e4.png)

Reply:
With DR6a I see the occasional worker killed, but it's not common.

With DR6c I see many workers getting killed and am trying to understand why.  My hope is that there is some configuration change that can be made to either the Dask Cluster setup or within the Dask Client or Notebook calls that help the workload be distributed more appropriately.

Reply:
> Was this with the suggested configuration of 32 total workers across two Haswell nodes?

Yes

Reply:
Cab you try using the fastparquet backend and see if it gets better? I saw huge memory usage in pyarrow compared to fastparqeut in some of my stress tests I did before.

Reply:
> Sorry for my unclear statement, I was talking about the options of NERSC jupyter hub:

Hmmm... I've always been running in "Shared CPU Node", and I've been able to talk to the workers fine using exactly the commands in the Notebook.

What was the error you got when running this way?

Was the output of the `client` cell as expected.  I.e., listed 32 workers and total memory and the Dask Dashboard?
 But then when you actually tried to run, your got messages that the workers were unavailable?

Reply:
> Can you try using the fastparquet backend and see if it gets better? I saw huge memory usage in pyarrow resulting in worker issues compared to fastparqeut in some of my stress tests I did before.

Thank you for continuing to wave the `fastparquet` banner.  I will add that to my testing.

Reply:
> Hmmm... I've always been running in "Shared CPU Node", and I've been able to talk to the workers fine using exactly the commands in the Notebook.
> 
> What was the error you got when running this way?
> 
> Was the output of the `client` cell as expected. I.e., listed 32 workers and total memory and the Dask Dashboard?
> But then when you actually tried to run, your got messages that the workers were unavailable?

The commands to set up dask workers were fine, but the notebook just hang at the `if not LOCAL_DASK:` cell. I then open a new notebook with exclusive jupyter node and then it works (I didn't even restart the dask workers). So I thought it was because the shared nodes cannot connect to the dask workers.

Reply:
> The commands to set up dask workers were fine, but the notebook just hang at the if not LOCAL_DASK: cell.

Was a SCHEDULER_FILE written to the right place?

Reply:
Yes, I checked that while the notebook hung. And the notebook ran in the exclusive note were able to pick it up.

Reply:
So I actually don't even have the option to choose between Shared and Exclusive.  I only get to pick Shared.  Is there something out-of-date with my configuration so that my Shared is different than your Shared?

Reply:
> Can you try using the fastparquet backend and see if it gets better? I saw huge memory usage in pyarrow resulting in worker issues compared to fastparqeut in some of my stress tests I did before.

In the immediate, this isn't instantly trivial to test on NERSC because our Parquet files are written out with SNAPPY compression and the available version of `fastparquet` in `desc-python` wasn't built with support for SNAPPY compression.

I've filed a ticket to add this and then will compare `pyarrow` and `fastparquet` in memory and speed performance.

Reply:
I actually don't know! CC @heather999 to see if Heather knows about this difference.